pages/implementation/start-using-avo-codegen.mdx

```diff
@@ -5,15 +5,59 @@ title: 'Start using Avo Codegen'
+
+## When use Codegen
+
+Codegen is our take on increasing the robustness of the analytics implementation and simplifying the implementation process.
+
+Adopting Codegen is not required to get value from Avo - many companies have their own ways to achieve robust analytics implementation and prefer to stick to them after adopting Avo and it’s totally fine.
+
+Avo is designed to be efficient and powerful without the Codegen, providing the option of the advanced tracking plan management with any implementation approach. 
+The decision to use Codegen is usually in the hands of the Engineering department. If they prefer to use a different approach Avo would help you communicate the tracking plan changes to the engineering team and help them to give the implementation feedback.
+
+## How does Codegen work
+
+Codegen works with any analytics tool, whether it's any of the third-party event logging SDKs (e.g. Google Analytics 4, Segment, RudderStack, mParticle, Snowplow, Amplitude, Mixpanel, Pendo, etc) or your homemade API.
+
+When designing the tracking plan you configure the list of analytics destinations for each source. In the generated code each destination is represented as its own Destination Interface in the initAvo method. This is how Codegen routes data, locally in your client (without the data ever going through Avo's servers), to the correct destination based on the source/destination connections you've defined in the Avo dashboard.
+
+In the Avo Tracking Plan you can configure which analytics <Link passHref href="/workspace/tracking-plan/events#a-nameactionsa-actions"><a>actions</a></Link> should be included in Codegen (for example you might want to update user properties or log revenue along with logging an event), and Codegen will handle calling all the relevant Destination Interface methods. All you have to do is call the generated function in your code base and pass in type safe key-value pairs, and the Codegen will take care of the rest.
+
+Learn more technical details about Codegen <Link passHref href="/implementation/avo-functions-overview"><a>here</a></Link>.
+
+## Do events flow through Avo servers?
+
+No. None of your events ever go through Avo servers. They go directly from your devices (whether that's a front-end client or a server) to the analytics destination.
+
+> 💡 This sometimes called "device mode". With CDPs, you can choose between <Link passHref href="https://segment.com/docs/guides/intro-impl/#connection-modes"><a>_Cloud-mode_ or _Device-mode_</a></Link>:
+>
+> - In _Cloud-mode_, your events go through the CDP servers. Your devices send the events to the CDP cloud (aka the CDP Servers) and the CDP cloud sends them to the analytics destinations.
+> - In _Device-mode_ the events go directly from your devices (whether it's a front end client or a server) to the analytics destination.
+>   In this regard, Avo works like _Device-mode_.
+
+## How to set up and use Codegen
```

Comment:
I'm wondering if we should split this doc in two somehow.

Feel like everything above is answering the question about what Codegen is, and everything below is about how to get started (quickstart guide)?

Reply:
Yes, I think it makes sense. 👍